src/enums.h

```diff
@@ -363,4 +363,26 @@ std::array<T, N> vec_to_array(const std::vector<T> &vec)
     return array;
 }
 
+template <typename T>
```

Comment:
Yes, this is definitely what `std::find` does, so use that. It has the advantage to work with all containers and iterators, even for a `std::array` (because the array exports `begin()` and `end()` iterators).

While this will work now, what about the future when somebody wants to use a `std::set` instead of a vector, or a `std::list`? This is all the same for `std::find` because all those containers have the `begin` and `end` functions.

`std::find` is also better because the type of the iterator is independent of the type of the value. I can search for a `const char*` in a `vector<string>` and it will work (with no implicit conversion).

It *may* be fine to add a tiny wrapper that handles the `begin()` and `end()` calls, like:
```C++
template<typename C, typename T>
bool is_any_of( const Container &c, const T &target ) {
    return std::find( c.begin(), c.end(), target ) != c.end();
}
```

Btw. functions in a header should be inline, otherwise they might be included in several compilation units and the linker will complain that they are defined multiple times.